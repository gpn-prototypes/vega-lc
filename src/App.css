html,
body,
#root {
  margin: 0;
  padding: 0;

  width: 100%;
  height: 100%;
}

.App {
  display: flex;
  flex-direction: column;
  align-items: flex-start;

  width: 100%;
  height: 100%;

  background: var(--color-bg-default);
}
