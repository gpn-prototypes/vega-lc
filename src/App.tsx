import React from 'react';
import { Route, Switch } from 'react-router-dom';
import { presetGpnDark, Theme } from '@gpn-prototypes/vega-ui';

import { GeologicalExploration } from './components/geological-exploration';
import Navigation from './components/navigation';

import './App.css';
import './styles/colors.css';

export const App = (): React.ReactElement => {
  return (
    <Theme className="App" preset={presetGpnDark}>
      <Navigation />
      <Switch>
        <Route exact component={GeologicalExploration} path="/" />
      </Switch>
    </Theme>
  );
};
