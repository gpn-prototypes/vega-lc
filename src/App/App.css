:root {
  --tooltip-z-index: 99;
}

* {
  box-sizing: border-box;
}

html,
body,
#root {
  margin: 0;
  padding: 0;

  width: 100%;
  height: 100%;
}

.LC__App-Wrapper {
  z-index: 0;

  display: flex;
  flex-direction: column;
  flex-grow: 1;

  width: 100%;
  height: 100%;
}

.LC__App {
  display: flex;
  overflow: hidden;
  flex-direction: column;
  flex-grow: 1;

  min-height: 100%;

  color: var(--color-typo-primary);
  background-color: var(--color-bg-default);
}

.Tooltip {
  z-index: var(--tooltip-z-index);
}

.SelectDropdown {
  z-index: 1;
}
