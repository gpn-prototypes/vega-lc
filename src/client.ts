import { ApolloClient, HttpLink, InMemoryCache } from '@apollo/client';
import fetch from 'cross-fetch';

import { authHeader } from './utils/set-auth-token';

const headers = {
  ...authHeader(),
};

export const mainLink = new HttpLink({ uri: 'graphql', headers, fetch });

export const projectLink = new HttpLink({
  uri: 'graphql/a3333333-b111-c111-d111-e00000000000',
  headers,
  fetch,
});

const client = new ApolloClient({
  link: mainLink,
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          logic: {
            merge: false,
          },
          domain: {
            merge: false,
          },
        },
      },
    },
  }),
});

export default client;
