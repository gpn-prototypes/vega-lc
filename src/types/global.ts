import ResizeObserver from 'resize-observer-polyfill';

declare global {
  interface Window {
    ResizeObserver: typeof ResizeObserver;
  }
}
