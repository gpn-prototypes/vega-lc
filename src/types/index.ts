export interface Identity {
  getToken(): Promise<string>;
}
