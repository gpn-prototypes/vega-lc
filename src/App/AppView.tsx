import React, { useContext, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Loader } from '@gpn-prototypes/vega-ui';

import { cnApp } from './cn-app';

import './App.css';
import '../styles/colors.css';

import { GeologicalExploration } from '@/components/geological-exploration';
import { ProjectErrors } from '@/components/project-errors';
import { ProjectContext } from '@/react-context/providers';
import { fetchVersion } from '@/redux-store/version/actions';
import { StoreLC } from '@/types/redux-store';

export const AppView = (): React.ReactElement => {
  const dispatch = useDispatch();

  const { initialized } = useContext(ProjectContext);

  const isLoading = useSelector<StoreLC, boolean>(
    (state) => !state.projectStructure.projectStructureQuery,
  );

  useEffect(() => {
    if (initialized) {
      dispatch(fetchVersion());
    }
  }, [dispatch, initialized]);

  return (
    <div className={cnApp('App')}>
      {isLoading ? <Loader size="m" /> : <GeologicalExploration />}
      <ProjectErrors />
    </div>
  );
};
