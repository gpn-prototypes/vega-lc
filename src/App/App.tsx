import React from 'react';
import { ApolloClient, NormalizedCacheObject } from '@apollo/client';
import { Root } from '@gpn-prototypes/vega-ui';

import { Providers } from '../components/providers';

import { AppView } from './AppView';
import classNames from './cn-app';

import './App.css';

interface AppProps {
  graphqlClient?: ApolloClient<NormalizedCacheObject>;
}

export const App: React.FC<AppProps> = (props) => {
  const { graphqlClient } = props;

  return (
    <Root
      initialPortals={[{ name: 'modalRoot' }]}
      defaultTheme="dark"
      className={classNames('App-Wrapper').toString()}
    >
      <Providers graphqlClient={graphqlClient}>
        <AppView />
      </Providers>
    </Root>
  );
};
