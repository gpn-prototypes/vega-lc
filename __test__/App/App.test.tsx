import React from 'react';
import renderer from 'react-test-renderer';
import { ApolloClient, InMemoryCache } from '@apollo/client';
import { render } from '@testing-library/react';
import ResizeObserver from 'resize-observer-polyfill';

import '../../src/types/global';

import { App } from '../../src/App/App';

jest.mock('../../src/App/AppView', () => {
  return {
    AppView: () => {
      return <div />;
    },
  };
});

const renderApp = () => {
  const mockGraphqlClient = new ApolloClient({ cache: new InMemoryCache() });
  const mockIdentity = {
    getToken: (): Promise<string> => new Promise((resolve) => setTimeout(() => resolve('token'))),
  };
  return render(<App graphqlClient={mockGraphqlClient} identity={mockIdentity} />);
};

describe('App', () => {
  test('рендерится без ошибок', () => {
    expect(renderApp).not.toThrow();
  });

  test('соответствует снапшоту', () => {
    window.ResizeObserver = ResizeObserver;

    const dom = renderer.create(<App />).toJSON();
    expect(dom).toMatchSnapshot();
  });
});
