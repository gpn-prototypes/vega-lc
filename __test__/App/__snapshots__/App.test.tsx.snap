// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App соответствует снапшоту 1`] = `
<div
  className="Theme Theme_color_gpnDark Theme_control_gpnDefault Theme_font_gpnDefault Theme_size_gpnDefault Theme_space_gpnDefault LC__App-Wrapper"
>
  <div />
</div>
`;
